### DISCUSSION TITLE
More items to the auto-shop #177

### DISCUSSION THREAD

**Discussion by Farigh** [Farigh/pokeclicker-automation#179] [PR discussions]
@Daniferrito I agree that this panel should allow more items to be automatically bought

However, there should be a list of targeted items, instead of filtering them.
The `Quest points` tab should be replaced with an `Eggs` one, only containing eggs
The `Farm points` tab should be renamed `Farm tools`

**Comment by Farigh** [comment]
Another thought: Some items have their price permanently increased after you by some (masterball, protein....)
Those should not display any `max base price`

**Comment by Daniferrito** [comment]
> However, there should be a list of targeted items, instead of filtering them.

So you want a whitelist instead of a blacklist.
My reasoning for implementing the blacklist is to better support future updates, so that the code here needs to be updated less when the game updates.

For example, if they add a new egg type, fairy, for example, we dont need to add it manually, or a new type of ball (that is not a complex trade)
On the other hand, if they add a new category of item, like megastones, we probably dont want them.

So I will just change it to a whitelist by type of items, allowing:
* Battle Items
* Energy restore
* Mulch
* Pokeballs (all types)
* Shovels
* Mulch shovels (they are different class from normal shovels)
* Vitamins
* Eggs

Item types that are NOT in the whitelist:
* Berries (but probably want to add some code to the farm to buy them when available if "unlock new berries" is on)
* Evolution Stones (but probably want to add some code to auto-buy them and use them in some future focus modes)
* Pokemon Items (but probably want to add some code to auto-buy them in some future focus modes)
* Farm Hands
* Flutes (but those are trades anyway)
* Hatchery helpers
* Held items (if the system gets reworked, maybe want to add them later, we will see)
* PokeBlocks (looks to be dead code, but might get fleshed out later)
* Key Items (but probably want to add some code to auto-buy them in some future focus modes)
* Oak Items (Same as key items)

The only thing that makes some kind of sense to add to the auto-shop are evolution stones, but I agree that there are too damn many.
The rest of the items are single-buy items anyway, so they make no sense here.

> The Quest points tab should be replaced with an Eggs one, only containing eggs

What about Beast balls? I agree that they are not a good mechanic, but we might want to allow to auto-buy them anyways. Adding them to the pokedollar part does not seem right, as they are sold for quest points, and the eggs section no longer fits them.
Maybe we could split the currency tabs with separators, so in the "quest points" tab the eggs, evo stones and pokeballs are clearly separated, and maybe even make the sections collapsable, so the evo stones do not take too much space (unless it is expanded)

**Reply by Farigh** [reply]
I would prefer to avoid adding it for now, and keep tabs with a bit of meaning instead of just currencies, as it would not have a spot that make sense, and as you rightly say are not that usefull as-is.

We might consider more tabs later if the needs presents itself

**Reply by Farigh** [reply]
I prefer a whitelist mainly for 2 reasons : 
- to avoid the menu to become less user-friendly
- to have tabs with meaning and easy to find by users

**Comment by Farigh** [comment]
Closing this as the PR was merged
